src/writer.jl

```diff
@@ -1,398 +1,277 @@
-#################################################################
-# This file implements a .pomdpx file generator using the
-# POMDPs.jl interface.
-#################################################################
-
 abstract type AbstractPOMDPXFile end
 
-mutable struct POMDPXFile <: AbstractPOMDPXFile
-    file_name::AbstractString
-    description::AbstractString
+@with_kw struct POMDPXFile <: AbstractPOMDPXFile
+    filename::String
+    description::String = "This is a POMDPX file for a POMDP"
 
-    state_name::AbstractString
-    action_name::AbstractString
-    reward_name::AbstractString
-    obs_name::AbstractString
+    s_var_name::String = "state"
+    a_var_name::String = "action"
+    o_var_name::String = "observation"
+    r_var_name::String = "reward"
 
-    initial_belief::Vector{Float64}
+    s_name::Function = normalize
+    a_name::Function = normalize
+    o_name::Function = normalize
 
-    #initial_belief::Vector{Float64} # belief over partially observed vars
+    pretty::Bool = false
+end
 
-    function POMDPXFile(file_name::AbstractString; description::AbstractString="",
-                    initial_belief::Vector{Float64}=Float64[])
+POMDPXFile(filename::String, pretty::Bool) = POMDPXFile(; filename=filename, pretty=pretty)
+POMDPXFile(filename::String) = POMDPXFile(filename, false)
```

Comment:
you can do this in one line
```suggestion
POMDPXFile(filename::String, pretty::Bool=true) = POMDPXFile(; filename=filename, pretty=pretty)
```